### PR TITLE
WIP - Membership type on checkout

### PR DESCRIPTION
**Pull request by pollygee**
Working on #260 
 Need some help!

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -36,4 +37,9 @@ def self.to_csv
   def send_welcome_email
     WelcomeMailer.welcome_email(self).deliver
   end
+
+  def name_with_membership_type
+    memb_type = self&.membership_type&.short_name || "Needs a membership type"
```
Comment (pollygee) [inline app/models/user.rb]: the first condition is always false for some reason.

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -22,6 +23,7 @@
         <td><%= user.phone_number %></td>
         <td><%= user.created_at %></td>
         <td><%= user.role.humanize %></td>
+<!--        <td><%#= MembershipType.find(user&.membership_type_id).name %></td>-->
```
Comment (pollygee) [inline app/views/users/index.html.erb]: I feel like here I should be able to say `user.membership_type.name` but for some reason I can't.  Is there anything obvious missing in the PR that makes this break?

File: db/migrate/20200206144913_add_membership_type_ref_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddMembershipTypeRefToUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_reference :users, :membership_type, foreign_key: true
```
Comment (pollygee) [inline db/migrate/20200206144913_add_membership_type_ref_to_users.rb]: Do I need this on the users table?

Comment (mark100net) [reply db/migrate/20200206144913_add_membership_type_ref_to_users.rb]: @pollygee  It kind of depends. If we put the membership type on the user's table, then we have to put this there, along with the expiration date and any other data that relates to the membership. A better alternative might be to have a memberships table (which would have `membership_type`, `user_id`, `expiration_date`, etc.). 

The remaining question would be whether the user `has_one` membership or `has_many` memberships (in the latter case we would add a new membership whenever a user joins or renews. The latter retains all membership history while being slightly more complicated to code.

Comment (pollygee) [reply db/migrate/20200206144913_add_membership_type_ref_to_users.rb]: I like the idea of having a memberships table.  Then we can see a history of the memberships as well.

**Comment by mark100net** [comment]
This is being handled elsewhere.

**Pull request closed without merging**